tools/babel-config.js

```diff
@@ -0,0 +1,21 @@
+// This is the same set of transformations that we will enable in a future
+// official React package
+module.exports = {
+  sourceMaps: "inline",
+  compact: false,
+  whitelist: [
+    'es6.arrowFunctions',
+    'es6.blockScoping',
+    'es6.classes',
+    'es6.destructuring',
+    'es6.parameters.rest',
+    'es6.properties.computed',
+    'es6.properties.shorthand',
+    'es6.spread',
+    'es6.templateLiterals',
+    'es7.trailingFunctionCommas',
+    'es7.objectRestSpread',
+    'flow',
```

Comment:
Babel's parser can parse flow annotations (i.e. colons followed by types).  The "flow" transformer simply deletes them, which is the support you need in your toolchain in addition to the Flow type-checker.  So it's definitely an option.